pkg/provider/onboardbase/client/client.go

```diff
@@ -0,0 +1,402 @@
+	}
+	_, err = c.performRequest("/secrets", "DELETE", headers{}, params, body)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *OnboardbaseClient) makeGetSecretsRequest(request SecretsRequest) (*secretResponseBody, *apiResponse, error) {
+	headers := headers{}
+
+	params := request.buildQueryParams()
+	response, apiErr := c.performRequest("/secrets", "GET", headers, params, httpRequestBody{})
+	if apiErr != nil {
+		return nil, nil, apiErr
+	}
+
+	var data *secretResponseBody
+	if err := json.Unmarshal(response.Body, &data); err != nil {
+		return nil, nil, &APIError{Err: err, Message: "unable to unmarshal secret payload", Data: string(response.Body)}
+	}
+	return data, response, nil
+}
+
+func (c *OnboardbaseClient) GetSecrets(request SecretsRequest) (*SecretsResponse, error) {
+	data, response, err := c.makeGetSecretsRequest(request)
+	if err != nil {
+		return nil, err
+	}
+
+	secrets, _ := c.getSecretsFromPayload(data.Data)
+	return &SecretsResponse{Secrets: secrets, Body: response.Body}, nil
+}
+
+func (r *SecretsRequest) buildQueryParams() queryParams {
+	params := queryParams{}
+
+	if r.Project != "" {
+		params["project"] = r.Project
+	}
+
+	if r.Environment != "" {
+		params["environment"] = r.Environment
+	}
+
+	return params
+}
+
+func (r *SecretRequest) buildQueryParams() queryParams {
+	params := queryParams{}
+
+	if r.Project != "" {
+		params["project"] = r.Project
+	}
+
+	if r.Environment != "" {
+		params["environment"] = r.Environment
+	}
+
+	return params
+}
+
+func (c *OnboardbaseClient) performRequest(path, method string, headers headers, params queryParams, body httpRequestBody) (*apiResponse, error) {
+	urlStr := c.BaseURL().String() + path
+	reqURL, err := url.Parse(urlStr)
+	if err != nil {
+		return nil, &APIError{Err: err, Message: fmt.Sprintf("invalid API URL: %s", urlStr)}
+	}
+
+	var bodyReader io.Reader
+	if body != nil {
+		bodyReader = bytes.NewReader(body)
+	} else {
+		bodyReader = http.NoBody
+	}
+
+	req, err := http.NewRequest(method, reqURL.String(), bodyReader)
+	if err != nil {
+		return nil, &APIError{Err: err, Message: "unable to form HTTP request"}
+	}
+
+	if method == "POST" && req.Header.Get("content-type") == "" {
+		req.Header.Set("content-type", "application/json")
+	}
+
+	if req.Header.Get("accept") == "" {
+		req.Header.Set("accept", "application/json")
+	}
+	req.Header.Set("user-agent", c.UserAgent)
+	req.Header.Set("api_key", c.OnboardbaseAPIKey)
+
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+
+	query := req.URL.Query()
+	for key, value := range params {
+		query.Add(key, value)
+	}
+	req.URL.RawQuery = query.Encode()
+
+	r, err := c.httpClient.Do(req)
+
+	if err != nil {
+		return nil, &APIError{Err: err, Message: "unable to load response"}
+	}
+	defer r.Body.Close()
+
+	bodyResponse, err := io.ReadAll(r.Body)
+	if err != nil {
+		return &apiResponse{HTTPResponse: r, Body: nil}, &APIError{Err: err, Message: "unable to read entire response body"}
+	}
```

Comment:
Interesting. But since if error is defined the other response is ignored, I don't see the point of `&apiResponse` here?

Reply:
So, it could return APIError instead?